manifests/client.pp

```diff
@@ -46,17 +46,14 @@
 # Copyright (C) 2012 Mike Arnold, unless otherwise noted.
 #
 class snmp::client (
-  $snmp_config        = $snmp::params::snmp_config,
-  $ensure             = $snmp::params::ensure,
-  $autoupgrade        = $snmp::params::safe_autoupgrade,
-  $package_name       = $snmp::params::client_package_name
+  $snmp_config         = $snmp::params::snmp_config,
+  $ensure              = $snmp::params::ensure,
```

Comment:
Want to make this an `Enum['present', 'absent']` while you're at it? It's fine to do that in another PR.

Reply:
Considering I randomly deleted a few lines just removing `validate_bool` *and* you approved it, probably best we keep with small PRs (and I get more sleep!) ;)